app/pages/sections/supplier/select/ui.test.js

```diff
@@ -0,0 +1,177 @@
+import nock from 'nock';
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import { solutionsApiUrl } from '../../../../config';
+import content from './manifest.json';
+
+const pageUrl = 'http://localhost:1234/organisation/order-1/supplier/search/select';
+
+const setCookies = ClientFunction(() => {
+  const cookieValue = JSON.stringify({
+    id: '88421113', name: 'Cool Dude', ordering: 'manage', primaryOrganisationId: 'org-id',
+  });
+
+  document.cookie = `fakeToken=${cookieValue}`;
+});
+
+const pageSetup = async (t, withAuth = false) => {
+  if (withAuth) {
+    await setCookies();
+  }
+};
+
+
+const getLocation = ClientFunction(() => document.location.href);
+
+fixture('Supplier select page')
+  .page('http://localhost:1234/some-fake-page')
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
+test('when user is not authenticated - should navigate to the identity server login page', async (t) => {
+  await pageSetup(t);
+  nock('http://identity-server')
+    .get('/login')
+    .reply(200);
+
+  await t.navigateTo(pageUrl);
+
+  await t
+    .expect(getLocation()).eql('http://identity-server/login');
+});
+
+test('should render Supplier select page', async (t) => {
+  nock(solutionsApiUrl)
+    .get('/api/v1/suppliers?name=some-supp')
+    .reply(200, [{}]);
+
+  await pageSetup(t, true);
+  await t.navigateTo(`${pageUrl}?supplierNameToFind=some-supp`);
+  const page = Selector('[data-test-id="supplier-select-page"]');
+
+  await t
+    .expect(page.exists).ok();
+});
+
+test('should navigate to /organisation/order-1/supplier/search when click on backlink', async (t) => {
+  nock(solutionsApiUrl)
+    .get('/api/v1/suppliers?name=some-supp')
+    .reply(200, [{}]);
+
+  await pageSetup(t, true);
+  await t.navigateTo(`${pageUrl}?supplierNameToFind=some-supp`);
```

Comment:
you have the pageUrl variable at the top, you could add another with the query string?

Reply:
again will be removed once we have state. OK to leave for now?

Reply:
global search and replace 😆